pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BasePartitionUpsertMetadataManager.java

```diff
@@ -1111,6 +1132,9 @@ protected void addDocId(IndexSegment segment, ThreadSafeMutableRoaringBitmap val
     try {
       doAddDocId(validDocIds, queryableDocIds, docId, recordInfo);
     } finally {
+      if (_enableSnapshot) {
```

Comment:
I'd suggest to move this if-block outside the critical section of upsertViewLock

btw, it may help to comment that we don't need to take snasphotLock when updating this Set (IIUC)

Reply:
> Actually this is not inside upsertViewLock. I just wanted to keep it in the finally block so that we add the updated segment to the set always.

Nvm! I will move this below after we do `_upsertViewLock.readLock().unlock();` if block.